### ISSUE TITLE
Metrics are blocking the event loop after update to version 3.9.14

### ISSUE THREAD

**Issue by cristian-joseb** [vert-x3/vertx-micrometer-metrics#169]
### Questions

Do not use this issue tracker to ask questions, instead use one of these [channels](https://vertx.io/community/). Questions will likely be closed without notice.

### Version

Which version(s) did you encounter this bug ?
3.9.14

### Context

When applying a filter to metric, and the same return Deny the event loop is blocked indefinitely. 

### Do you have a reproducer?
https://github.com/cristian-joseb/vertx-metrics

A reproducer is a simple project hosted on GitHub (or another forge supporting git clone operation) that has a build file that can be executed to reproduce the issue.

Reproducers are very helpful for contributors and will likely help them fixing your bug faster.

* Link to github project/gist
https://github.com/cristian-joseb/vertx-metrics

### Steps to reproduce

1.  clone https://github.com/cristian-joseb/vertx-metrics
2.  open with an IDE and run Sever.java
3.  go the browser and paste http://localhost:8080/ReadStatus.v1
4.  the first metrics related to ReadStatus.v1 will be recorded
5.  the second metric related to http://localhost:8080/favicon.ico  will cause the issue.

### Extra

* Anything that can be relevant such as OS version, JVM version

**Comment by cristian-joseb** [comment]
The problem seem to be the Gauge class get method implementation, the below code used to work 
```public synchronized T get(String... values) {
    // This method is synchronized to make sure the "T" built via supplier will match the one passed to Gauge
    //  since it is stored as WeakReference in Micrometer DefaultGauge, it must not be lost.
    T candidate = tSupplier.get();
    ToDoubleFunc<T> candidateFunc = new ToDoubleFunc<>(dGetter);
    // Register this gauge if necessary
    // Note: we need here to go through the process of Gauge creation, even if it already exists,
    //  in order to get the Gauge ID. This ID generation is not trivial since it may involves attached MetricFilters.
    //  Micrometer will not register the gauge twice if it was already created.
    Gauge gauge = Gauge.builder(name, candidate, candidateFunc)
      .description(description)
      .tags(Labels.toTags(keys, values))
      .register(registry);
    Meter.Id gaugeId = gauge.getId();
    Object res = gauges.computeIfAbsent(gaugeId, v -> candidate);
    return (T) res;
  }```

in the new implementation  the below code is not executing when the filter return DENY
```
if (candidateFunc.invoked) {
        gauges.put(gaugeId, candidate);
        res = candidate;
        break;
      }
```

and hence it is going on a infinite loop.

**Comment by cristian-joseb** [comment]
@tsegismont PR https://github.com/vert-x3/vertx-micrometer-metrics/pull/170

**Comment by tsegismont** [comment]
This has been reported in #150 , but the fix has not been backported to 3.9

**Comment by tsegismont** [comment]
Fixed in #171
